src/utils/regex.cc

```diff
@@ -311,5 +378,43 @@ int Regex::search(const std::string& s) const {
 #endif
 }
 
+unsigned long Regex::get_default_match_limit() const {
```

Comment:
I'm not sure I see the value in doing this.

If the ModSecurity config has not specified a limit, why do we need to do anything special here?

Why not just allow the builtin limits within PCRE to take effect?

I'm a little bit concerned especially that this code will get run for every attempt at a match -- even a relatively small number of cycles and memory jumps can add up.